extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/stork/StorkIntegrationTest.java

```diff
@@ -15,54 +15,65 @@
 import org.junit.jupiter.api.Timeout;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-import io.quarkus.rest.client.reactive.HelloClient2;
-import io.quarkus.rest.client.reactive.HelloResource;
 import io.quarkus.test.QuarkusUnitTest;
 import io.smallrye.stork.api.NoSuchServiceDefinitionException;
 
 public class StorkIntegrationTest {
     @RegisterExtension
     static final QuarkusUnitTest TEST = new QuarkusUnitTest()
             .withApplicationRoot((jar) -> jar
-                    .addClasses(HelloClient2.class, HelloResource.class))
+                    .addClasses(HelloClient.class, HelloResource.class))
             .withConfigurationResource("stork-application.properties");
 
     @RestClient
-    HelloClient2 client;
+    HelloClient client;
 
     @Test
     void shouldDetermineUrlViaStork() {
         String greeting = RestClientBuilder.newBuilder().baseUri(URI.create("stork://hello-service/hello"))
-                .build(HelloClient2.class)
+                .build(HelloClient.class)
                 .echo("black and white bird");
         assertThat(greeting).isEqualTo("hello, black and white bird");
+
+        greeting = RestClientBuilder.newBuilder().baseUri(URI.create("stork://hello-service/hello"))
+                .build(HelloClient.class)
+                .helloWithPathParam("black and white bird");
+        assertThat(greeting).isEqualTo("Hello, black%20and%20white%20bird");
     }
 
     @Test
     void shouldDetermineUrlViaStorkWhenUsingTarget() throws URISyntaxException {
-        String greeting = ClientBuilder.newClient().target("stork://hello-service/hello").request().get(String.class);
-        assertThat(greeting).isEqualTo("Hello");
+        String greeting = ClientBuilder.newClient().target("stork://hello-service/hello").request()
+                .get(String.class);
+        assertThat(greeting).isEqualTo("Hello, World!");
 
         greeting = ClientBuilder.newClient().target(new URI("stork://hello-service/hello")).request().get(String.class);
-        assertThat(greeting).isEqualTo("Hello");
+        assertThat(greeting).isEqualTo("Hello, World!");
 
         greeting = ClientBuilder.newClient().target(UriBuilder.fromUri("stork://hello-service/hello")).request()
                 .get(String.class);
-        assertThat(greeting).isEqualTo("Hello");
+        assertThat(greeting).isEqualTo("Hello, World!");
+
+        greeting = ClientBuilder.newClient().target("stork://hello-service/hello").path("big bird").request()
+                .get(String.class);
+        assertThat(greeting).isEqualTo("Hello, big%20bird");
```

Comment:
Why it's not decoded here?

Reply:
Resteasy encodes the request's pathParam containing a blank into big%20bird. Stork appends this in the new URI once it knows the actual endpoint. The remote resource receives that (encoded) String `big%20bird`and returns it in the response. The SyncInvoker#get method doesn't decode it so we have the String as it is.

Reply:
@geoand is it the expected behavior? I'm surprised that RestEasy Reactive does not decode the path parameter.

Reply:
I don't think it's correct. The parameter should be decoded

Reply:
The current issue is that the Stork filter is currently sending a raw path, thus an encoded big%20bird. The resource receives it so in the path param. How would the resource know that it is a string to be decoded? The encoding/decoding should already have happened. The resource, perceiving the string as it is without the need for decoding, sends it back.

Reply:
Okay, in that case it makes sense